avm/src/vm/run.rs

```diff
@@ -38,14 +38,14 @@ impl VM {
     }
   }
 
-  fn sched_event(self: &mut VM, event: EventEmit) {
+  fn sched_event(self: &mut VM, mut event: EventEmit) {
     // schedule 1st fragment of each handler of this event
     let handlers = Program::global().event_handlers.get(&event.id).unwrap();
     for (i, hand) in handlers.iter().enumerate() {
       // first fragment of this handler
       let frag = HandlerFragment::new(event.id, i);
       // memory frag representing the memory for each handler call
-      let hand_mem = HandlerMemory::new(event.payload.clone(), hand.mem_req);
+      let hand_mem = HandlerMemory::new(event.payload.take(), hand.mem_req);
```

Comment:
I'm confused because [take](https://doc.rust-lang.org/std/option/enum.Option.html#method.take) clears the event payload and sets it to `None` so the next handler that comes along

Reply:
ah... this will take a bit more to figure out then, since using `clone` increments the refcount of the payload...

Reply:
oof that sucks. I think subsequent events are payload-less. However we can verify with a bdd test that has a custom event with a payload and it register 2 handlers to it and checks that the payload is present